### DISCUSSION TITLE
nosql exporter?

### DISCUSSION THREAD

**Discussion by Lusitaniae** [burningalchemist/sql_exporter#274] [Ideas]
**Is your feature request related to a problem? Please describe.**
Would be nice to have support for nosql databases too (not necessarily in this repo/project)

**Describe the solution you'd like**
Support for nosql databases: mongo, redis,memcached, dynamodb, bigtable, etc

**Describe alternatives you've considered**

**Additional context**

**Comment by burningalchemist** [comment]
Hi @Lusitaniae, thank you for your suggestion! :) 

I think it might be tricky since SQL has a common interface whereas NoSQL databases usually have their own implementations, protocols, etc.

I found one abstraction library, which potentially might help in solving this problem (although we still won't have any common query language unless we design one) - [philippgille/gokv](https://github.com/philippgille/gokv). It doesn't seem to be actively maintained, though.

Since it's not an issue, and it won't become a part of SQL Exporter due to the domain difference, I'll move it to Discussions/Ideas section for the time being. 👍
